docs/segmentedtickerplant.md

```diff
@@ -0,0 +1,340 @@
+- endofday[date;data]
+  Called at the end of day for all modes. Takes 2 arguments date (current date) and data (a table containing some basic information on the stp process and the current time on the stp)
+
+**Error Trapping**
+
+If the `.stplg.errmode` Boolean variable is set to true, an error log is opened on start up and the `.u.upd` function is wrapped in an error trap, so that if a bad message is received, it is not published but instead sent to the error log. The advantage of this is that bad updates are not sent through or replayed into the subscribers, which could cause issues, and they are easier to find and debug.
+
+This mode is really designed for development/testing purposes, it shouldn't be necessary for a stable production feed and will add a small overhead to each update.
+
+**Time Zone Behaviour**
+
+A key tickerplant function is to timestamp the incoming data before it gets published to the rest of the system. In order to maintain applications and data in different time zones, such as a UK-based application which processes US data, the STP works with the `.eodtime` namespace variables in much the same way as other TorQ processes. The key variable used is `.eodtime.dailyadj` but more information on setting up a TorQ process in a different time zone can be found [here](https://aquaqanalytics.github.io/TorQ/utilities/#eodtimeq).
+
+**Customisation and Flexibility**
+
+The STP has been designed with customisation in mind. To this end here are a couple of ways to tailor the process to suit a particular application. The first is utilising the fact that each table has its own UPD function, meaning that some additional processing, such as adding a sequence number or a time-zone offset, can be done in the STP itself rather than needing to be done in a separate process. This is done by altering the `.stplg.updtab` dictionary in the segmentedtickerplant settings config file:
+
+```q
+// In file $TORQHOME/appconfig/settings/segmentedtickerplant.q
+// Apply a sequence number to 'tabname'
+.stplg.updtab[`tabname]:{((count first x)#'(y;.stplg.seqnum),x}
+
+// In the STP process
+q) .stplg.updtab
+quote   | {(enlist(count first x)#y),x}
+trade   | {(enlist(count first x)#y),x}
+tabname | {((count first x)#'(y;.stplg.seqnum),x}
+...
+```
+
+A second way that the process can be tailored is to make custom batching modes. The batching behaviour depends on two functions: `.u.upd` and `.z.ts`. The former is called every time an update arrives in the STP and the latter whenever the timer function is called (eg. if the process is started with `-t 1000`, this will be called every second). In the default batching mode, `.u.upd` inserts the update into a local table and writes it to the log file, and `.z.ts` publishes the contents of the local table before clearing it. To customise these functions, the `.stplg.upd` and `.stplg.zts` dictionaries will need to be customised. For example, the default batching code looks like the following:
```

Comment:
```suggestion
The batching behaviour depends on two functions: `.u.upd` and `.z.ts`. The former is called every time an update arrives in the STP and the latter whenever the timer function is called (e.g. if the process is started with `-t 1000`, this will be called every second). In the default batching mode, `.u.upd` inserts the update into a local table and writes it to the log file, and `.z.ts` publishes the contents of the local table before clearing it. To customise these functions, the `.stplg.upd` and `.stplg.zts` dictionaries will need to be customised. For example, the default batching code looks like the following:
```